### PR TITLE
[HIVEMALL-84-3] Update docker-compose.yml and documents

### PR DESCRIPTION
**Pull request by takuti**
## What changes were proposed in this pull request?

I tried things written in the Docker on Hivemall documentation and noticed a couple of things we can improve.

- Update `docker-compose.yml` to behave in a similar manner to `docker run`
- Update Docker on Hivemall documentation

## What type of PR is it?

Documentation

## What is the Jira issue?

https://issues.apache.org/jira/browse/HIVEMALL-84

## How was this patch tested?

Manually

### CODE REVIEW

**Comment by takuti** [comment]
@amaya382 could you check this if you get a chance?

File: docs/gitbook/docker/getting_started.md [hunk 1]
```diff
@@ -31,38 +31,73 @@ This page introduces how to run Hivemall on Docker.
...
+## Using `docker` command
   
-  `docker build -f resources/docker/Dockerfile .`
+```
+$ docker build -f resources/docker/Dockerfile .
+```
 
 > #### Note
-> You can [skip](./getting_started.html#running-pre-built-docker-image-in-dockerhub) building images by using existing Docker images.
+> You can [skip](./getting_started.html#running-pre-built-docker-image-in-dockerhub) building images by using a pre-build docker image from Docker Hub.
```
Comment (amaya382) [inline docs/gitbook/docker/getting_started.md]: `...-in-docker-hub` ?

**Comment by takuti** [comment]
@amaya382 Fixed document, thanks!

File: docs/gitbook/docker/getting_started.md [hunk 2]
```diff
@@ -31,38 +31,73 @@ This page introduces how to run Hivemall on Docker.
...
+
+```
+$ docker run ... --volume /path/to/local/hivemall:/opt/hivemall
+```
+
+## Running pre-built Docker image in Docker Hub
 
 You can find pre-built Hivemall docker images in [this repository](https://hub.docker.com/r/hivemall/latest/).
 
+1. Check [the latest tag](https://hub.docker.com/r/hivemall/latest/tags/) first
```
Comment (takuti) [inline docs/gitbook/docker/getting_started.md]: Ah, I see. This part is written by @myui in commit 10e7d450fa8257efc5d614957fda514b2b91fdee. 

Hi @myui, could you make further discussion in here or another issue? I guess this part was temporarily necessary for the BoF session of the Apache BigData conf.

Comment (takuti) [inline docs/gitbook/docker/getting_started.md]: Or, if @amaya382 can set up this, it would be nice after this PR has been merged.

Comment (myui) [inline docs/gitbook/docker/getting_started.md]: owner is used for repository.
repository is used for branch.
tag is used for versioning.

hivemall:release:v0.5-rc.1
hivemall:lastest:20170502-1
hivemall:spark:20170501

owner: hivemall is used only for hivemall. So, okey as it is.

Comment (amaya382) [inline docs/gitbook/docker/getting_started.md]: It's redundant but depart from docker's usual practice...
e.g.
* automated build uses branch or tag (on git) as tag (on docker)
* if become an official repository or under apache, its name will be broken

Comment (myui) [inline docs/gitbook/docker/getting_started.md]: I'll ask for ASF INFRA to get an access to  https://hub.docker.com/u/apache/

Comment (myui) [inline docs/gitbook/docker/getting_started.md]: @amaya382 @takuti official Dockerhub can only be used for official Apache releases.

See discussions in 
https://issues.apache.org/jira/browse/INFRA-14123?jql=project%20%3D%20INFRA%20AND%20text%20~%20dockerhub

Unofficial docker images better not be advertised officially...

Comment (amaya382) [inline docs/gitbook/docker/getting_started.md]: It seems better to distribute TAGGED images on docker hub (`apache/hivemall`) by automated build and distributing NOT TAGGED ones should depend on manual build from Dockerfile.
(The former for evaluation by user, the latter for developer)

http://www.apache.org/dev/release-distribution.html#unreleased

Comment (myui) [inline docs/gitbook/docker/getting_started.md]: We can only publish TAGGED image associated with ASF official release. CI release is not allowed in ASF distribution policies.

Comment (takuti) [inline docs/gitbook/docker/getting_started.md]: Even only for release, it sounds nice. Let's try it in the next (first) apache release if possible :)

For now, I've updated the description of the Docker Hub image. I keep the part, but mentioned that the distributed image might be out-of-data. Is it okay for you @myui @amaya382 ?

In terms of naming, it should be re-considered when we officially distribute an image according to ASF release.

**Comment by takuti** [comment]
Even only for release, it sounds nice. Let's try it in the next (first) apache release if possible :)

For now, I've updated the description of the Docker Hub image. I keep the part, but mentioned that the distributed image might be out-of-data. Is it okay for you @myui @amaya382 ?

In terms of naming, it should be re-considered when we officially distribute an image according to ASF release.

**Comment by amaya382** [comment]
@takuti looks good!
Let's reconsider in other issue/PR

**Comment by amaya382** [comment]
Created https://issues.apache.org/jira/browse/HIVEMALL-106

**Comment by takuti** [comment]
@myui ^ everything is okay?

**Comment by myui** [comment]
will check soon.

**Comment by myui** [comment]
LGTM. Merged to master. Thanks!

**Pull request closed without merging**